### ISSUE TITLE
Improve the Join community flow to only wait for the Join Permission response

### ISSUE THREAD

**Issue by jrainville** [status-im/status-desktop#14680]
### Description

Similar to what was done [here](https://github.com/status-im/status-desktop/commit/6b01af86c87aef5a5ab193f4543b904449fdd101#r141857471), but do it without the bug I introduced :see_no_evil: 

Currently, the join flow waits for all permission checks to finish (join permission and all channel permissions).

However, the channel permission checks are not necessary to join. Those checks are only useful when you go in the Select Shared Addresses view, to see which channel you'll have access to.

So basically, as soon as the check for the permission to join returns, we can update the button. If the permission to join is satisfied, the button becomes active.
If it is not satisfied, the button is disabled with the reason.

Also, to make sure someone doesn't make the same mistake as me, we should rename `joinPermissionsCheckSuccessful` to `joinPermissionsCheckFinished`

**Comment by jrainville** [comment]
@noeliaSD it would be very good to have this improvement in 2.29 if someone from your team has time. I'm guessing @caybro would be the best to take this on.

**Comment by caybro** [comment]
> @noeliaSD it would be very good to have this improvement in 2.29 if someone from your team has time. I'm guessing @caybro would be the best to take this on.

I can definitely do the UI part, but I guess it needs the relevant backend part first, similar to what @MishkaRogachev did before

**Comment by caybro** [comment]
@jrainville @MishkaRogachev are you guys gonna take care of the needed backend, or should I get my hands dirty? :) Second question, who's taking care of the delta we have now between 2.28.1 and master?

**Comment by jrainville** [comment]
> @jrainville @MishkaRogachev are you guys gonna take care of the needed backend, or should I get my hands dirty? 

Hopefully the backend already gives you the right info, since we already return some info when the join permission check returns (in edit shared addresses, we can see the Join permission becoming a checkmark when it finishes while the channel ones are still running).
If you do need some new backend properties, let us know and if we have time, we'll try to do it or at least give pointers.

> who's taking care of the delta we have now between 2.28.1 and master?

We can leave 2.28.1 as is. Currently, we don't let users create new communities (easily) and our Status community is not using a permission to join. So you can just do the fix on master.

**Comment by caybro** [comment]
> @jrainville @MishkaRogachev are you guys gonna take care of the needed backend, or should I get my hands dirty?\n\nHopefully the backend already gives you the right info, since we already return some info when the join permission check returns (in edit shared addresses, we can see the Join permission

Not quite but I guess I'll be able to figure this out. Currently the boolean returns only when both join and channel permissions check has finished, I need to separate that into two properties since we're only interested in the join permission only here.
